### PR TITLE
TASK: Add easy docker compose setup to get up and running quickly

### PR DESCRIPTION
**Pull request by JamesAlias**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,11 @@
 Build/Resources/Java/selenium-server-standalone*
 /"
 Upgrading.rst
+
+# Dev Settings
+!/Configuration/Development
+/Configuration/Development/*
+!/Configuration/Development/Settings.yaml
```
Comment (JamesAlias) [inline .gitignore]: Should this default `Settings.yaml` reside in `/Configuration/Development/Docker`?

File: Configuration/Development/Settings.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#        sqlLogger: 'Neos\Flow\Persistence\Doctrine\Logging\SqlLogger'
+
+    # If you are running Flow within a Docker environment, you probably need to allow all proxies,
+    # because the container acts as such and the IP is variable.
+#    http:
+#      trustedProxies:
+#        proxies: '*'
+
+  Imagine:
+    driver: Gmagick
```
Comment (JamesAlias) [inline Configuration/Development/Settings.yaml]: Is `Gmagick` the current "best practice"?

File: .gitignore (hunk 1 above)
Comment (JamesAlias) [reply .gitignore]: done

File: Configuration/Development/Docker/Settings.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    # because the container acts as such and the IP is variable.
+#    http:
+#      trustedProxies:
+#        proxies: '*'
+
+  Imagine:
+    driver: Gmagick
+
+
+Sandstorm:
```
Comment (lorenzulrich) [inline Configuration/Development/Docker/Settings.yaml]: This does not really belong here

File: Readme.rst [hunk 4]
```diff
@@ -11,6 +11,14 @@ Learn more about the Neos content application platform on http://www.neos.io/.
 Quick Start
 ===========
 
+Docker Setup
+------------
+
+First time will take some time because we load all neos/* dependencies as git repositories.
```
Comment (lorenzulrich) [inline Readme.rst]: `Git` in uppercase

**Comment by lorenzulrich** [comment]
> TODO:
> 
> * also add `ddev` setup?

I'd be in favour of having a ddev setup, yes. ddev is very easy to work with, widely supported and with good support. We're using ddev for all our development.

File: Configuration/Development/Docker/Settings.yaml (hunk 3 above)
Comment (JamesAlias) [reply Configuration/Development/Docker/Settings.yaml]: my bad, I already used the new docker setup to work 🙈

**Comment by JamesAlias** [comment]
> > TODO:
> > 
> > * also add `ddev` setup?
> 
> I'd be in favour of having a ddev setup, yes. ddev is very easy to work with, widely supported and with good support. We're using ddev for all our development.

Feel free to add it in a new PR 🙂 
I want to enable people as fast as possible and docker compose should be a reasonable alternative for everyone.

I know @mficzel uses ddev as well so get in touch if you need help or discussion.

**Comment by ClaraRuna** [comment]
DAU proof! Works great :)

**Comment by kdambekalns** [comment]
As this is merged, it is vital to adjust https://docs.neos.io/guide/installation-development-setup/docker-and-docker-compose-setup

**Comment by Kleisli** [comment]
> I'd be in favour of having a ddev setup, yes. ddev is very easy to work with, widely supported and with good support. We're using ddev for all our development.

ddev is definitely much easier than docker, I always ask myself who is "recommending" docker setup on docs.neos.io

but anyway, I have this work-in-progress Repo to setup Neos with a Site package on ddev, I'll probably add Carbon.Pipeline as a dependency for the frontend building..
https://github.com/Kleisli/neos-quickstart

**Comment by JamesAlias** [comment]
> As this is merged, it is vital to adjust https://docs.neos.io/guide/installation-development-setup/docker-and-docker-compose-setup @kdambekalns 

I talked with @tobiasgruber about it and we agreed to let this part of the docs untouched as it felt like the documentation to get startet with setting up a Neos _Base Distribution_ environment.

This setup is scoped to the Neos _Development Distribution_ and aimed at contributers.

The "entrypoint" to the "Getting Started" docs are at https://docs.neos.io/guide/installation-development-setup (see "Contributing" Hint). And the docs for the new dev setup are here: https://docs.neos.io/guide/contributing-to-neos/neos-core#docker-setup

**Comment by kdambekalns** [comment]
> ddev is definitely much easier than docker

Very much depends… I have had to work with ddev setups that didn't even start the way it was supposed to work. With a working `docker-compose.yaml` in place "plain Docker" can be as easy as `docker compose up`, too. 😎

**Comment by JamesAlias** [comment]
I will link docs urls on PRs in the future 🙂

**Comment by kdambekalns** [comment]
> This setup is scoped to the Neos _Development Distribution_ and aimed at contributers.

Ah, I see. Ok, I guess that is useful. Not sure the distinction is clear enough, but that's not your fault by any means. Could even just be me. 🤷‍♂️😇

**Comment by JamesAlias** [comment]
@Kleisli I agree with @kdambekalns.

There is nothing stopping you from adding a ddev setup and opening up a PR 🙂 
It's mentioned in the docs as well: https://docs.neos.io/guide/installation-development-setup/ddev-local-beach

**Comment by Kleisli** [comment]
> With a working `docker-compose.yaml` in place "plain Docker" can be as easy as `docker compose up`, too. sunglasses

I tend to agree on that, but the problem is how to get there.. ;-)

Having a `docker-compose.yaml` in the distribution will definitely make it easier, but I remember having a hard time defining and linking all the containers from scratch years ago, and with ddev I had to configure much less. And I can run multiple projects at the same time on my machine. :-D

While it makes it a bit more complicated for newbies to decide, it's always good to have options. :) I'll try to check what it needs to add a default ddev config to the distribution

**Pull request merged**